File: src/main/java/org/radarcns/management/security/jwt/algorithm/RsaJwtAlgorithm.java
package org.radarcns.management.security.jwt.algorithm;

import com.auth0.jwt.algorithms.Algorithm;

import java.security.KeyPair;
import java.security.interfaces.RSAPrivateKey;
import java.security.interfaces.RSAPublicKey;

public class RsaJwtAlgorithm extends AsymmetricalJwtAlgorithm {
    /** RSA JWT algorithm. */
    public RsaJwtAlgorithm(KeyPair keyPair) {
        super(keyPair);
        if (!(keyPair.getPrivate() instanceof RSAPrivateKey)) {
            throw new IllegalArgumentException(
                    "Cannot make RsaJwtAlgorithm with " + keyPair.getPrivate().getClass());
        }
    }

    @Override
    public Algorithm getAlgorithm() {
        return Algorithm.RSA256(
                (RSAPublicKey)keyPair.getPublic(),
                (RSAPrivateKey)keyPair.getPrivate());
    }

    @Override
    public String getEncodedStringHeader() {
        return "-----BEGIN PUBLIC KEY-----";
    }

    @Override
    public String getEncodedStringFooter() {
        return "-----END PUBLIC KEY-----";
    }
}

File: src/main/java/org/radarbase/management/security/jwt/algorithm/RsaJwtAlgorithm.kt
package org.radarbase.management.security.jwt.algorithm

import com.auth0.jwt.algorithms.Algorithm
import java.security.KeyPair
import java.security.interfaces.RSAPrivateKey
import java.security.interfaces.RSAPublicKey

class RsaJwtAlgorithm(keyPair: KeyPair) : AsymmetricalJwtAlgorithm(keyPair) {
    /** RSA JWT algorithm.  */
    init {
        require(keyPair.private is RSAPrivateKey) { "Cannot make RsaJwtAlgorithm with " + keyPair.private.javaClass }
    }

    override val algorithm: Algorithm
        get() = Algorithm.RSA256(
            keyPair.public as RSAPublicKey,
            keyPair.private as RSAPrivateKey
        )
    override val encodedStringHeader: String
        get() = "-----BEGIN PUBLIC KEY-----"
    override val encodedStringFooter: String
        get() = "-----END PUBLIC KEY-----"
    override val keyType: String
        get() = "RSA"
}
